pyro/contrib/forecast/forecaster.py

```diff
@@ -283,3 +284,80 @@ def forward(self, data, covariates, num_samples):
             with PrefixConditionMessenger(self.model._prefix_condition_data):
                 with pyro.plate("particles", num_samples, dim=dim):
                     return self.model(data, covariates)
+
+
+class HMCForecaster(nn.Module):
+    """
+    Forecaster for a :class:`ForecastingModel` using Hamiltonian Monte Carlo.
+
+    On initialization, this will run :class:`~pyro.infer.mcmc.nuts.NUTS` sampler
+    to get posterior samples of the model.
+
+    After construction, this can be called to generate sample forecasts.
+
+    :param ForecastingModel model: A forecasting model subclass instance.
+    :param data: A tensor dataset with time dimension -2.
+    :type data: ~torch.Tensor
+    :param covariates: A tensor of covariates with time dimension -2.
+        For models not using covariates, pass a shaped empty tensor
+        ``torch.empty(duration, 0)``.
+    :type covariates: ~torch.Tensor
+
+    :param int num_warmup: number of MCMC warmup steps.
+    :param int num_samples: number of MCMC samples.
+    :param int num_chains: number of parallel MCMC chains.
+    :param bool dense_mass: a flag to decide if mass matrix is dense or diagonal.
```

Comment:
nit: A flag that controls whether the mass matrix is dense or diagonal. Defaults to False.